gitlab/datadog_checks/gitlab/gitlab_v2.py

```diff
@@ -64,7 +65,12 @@ def get_default_config(self):
         }
 
     def create_scraper(self, config):
-        return OpenMetricsCompatibilityScraper(self, ChainMap({"tags": self._tags}, config, self.get_default_config()))
+        scraper = OpenMetricsCompatibilityScraper(
+            self, ChainMap({"tags": self._tags}, config, self.get_default_config())
+        )
+        if config['openmetrics_endpoint'] == config.get('gitaly_server_endpoint'):
+            scraper.SERVICE_CHECK_HEALTH = f"gitaly.{scraper.SERVICE_CHECK_HEALTH}"
```

Comment:
To reviewer: I did not find a better way to handle this. I tried to add a custom prefix but we are calling directly [this](https://github.com/DataDog/integrations-core/blob/96ceba2baffbeb0250849e62f66e00a8ac7d195b/datadog_checks_base/datadog_checks/base/checks/base.py#L1023).

Reply:
hmm, so there's a few things confusing to me:

1. when and why would `gitaly_server_endpoint` be the same as `openmetrics_endpoint`? Like, why would a customer set both these fields to the same value? Or is this the result of [this line](https://github.com/DataDog/integrations-core/pull/14316/files#diff-2439d42dadfb00b4237eceefcc649359d3b8faa3fbac51132a803a1865904090R132)?
2. Why aren't we adding a separate scraper for Gitaly at all?
3. I don't see the connection between `scraper.SERVICE_CHECK_HEALTH` and the method you linked in your comment, sorry

Reply:
1. The scraper are using the `openmetrics_endpoint` parameter to scrape the metrics. Customers do not set these options to the same values. What we are doing [here](https://github.com/DataDog/integrations-core/pull/14316/files#diff-2439d42dadfb00b4237eceefcc649359d3b8faa3fbac51132a803a1865904090R128-R134) is that if the customer set the `gitaly_server_endpoint`, we configure an additional scraper to the check to scrape the gitaly endpoint. This is a separate scraper. The problem is for the service check because to format the service check name we are using `SERVICE_CHECK_HEALTH` which is set [here](https://github.com/DataDog/integrations-core/blob/a8b01ea5ab8d2528c9f9f44d5ac30db734a0eb5f/datadog_checks_base/datadog_checks/base/checks/openmetrics/v2/scraper.py#L49)
2. See 1.
3. See [this](https://github.com/DataDog/integrations-core/blob/96ceba2baffbeb0250849e62f66e00a8ac7d195b/datadog_checks_base/datadog_checks/base/checks/base.py#L835-L857). The name of the service check for an openmetrics scraper is retrieved from scraper.SERVICE_CHECK_HEALTH

Happy to chat over zoom if you want

Reply:
As discussed over zoom, I just added some comments